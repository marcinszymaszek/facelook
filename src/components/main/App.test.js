import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import renderer from 'react-test-renderer';

import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('App tests', () => {

    it("App renders without a problem", () => {
        const div = document.createElement('div');
        ReactDOM.render(<App />, div);
        ReactDOM.unmountComponentAtNode(div);
    })

    it('Snapshot matches', () => {
        const tree = renderer.create(<App />);
        expect(tree).toMatchSnapshot();
    })

    it("Child components renders", () => {
        const wrapper = shallow(<App />);
        // console.log(wrapper.debug());
        expect(wrapper.find('MainInfo').exists()).toBe(true);
        expect(wrapper.find('Friends').exists()).toBe(true);
        expect(wrapper.find('Opinion').exists()).toBe(true);
        expect(wrapper.find('Users').exists()).toBe(true);
    })

})

