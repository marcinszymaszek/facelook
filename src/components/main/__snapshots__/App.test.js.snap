// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App tests Snapshot matches 1`] = `
Array [
  <div
    className="main-container"
  >
    <div
      className="inline-container"
    >
      <div
        className="inner-box"
      >
        <h3>
          Amount of users: 
        </h3>
        <svg
          className="eye-icon"
          onClick={[Function]}
        >
          eye-slash-solid.svg
        </svg>
      </div>
    </div>
    <div
      className="inline-container"
    >
      <div
        className="inner-box"
      >
        <h3>
          Your friends: 
          0
        </h3>
        <svg
          className="minus-circle-icon"
          onClick={[Function]}
        >
          minus-circle-solid.svg
        </svg>
        <svg
          className="plus-circle-icon"
          onClick={[Function]}
        >
          plus-circle-solid.svg
        </svg>
      </div>
    </div>
    <div
      className="inline-container"
    >
      <div
        className="inner-box"
      >
        <h3>
          Do you like it?
        </h3>
        <svg
          className="thumb-icon"
          onClick={[Function]}
        >
          thumbs-up-solid.svg
        </svg>
      </div>
    </div>
    <div
      className="inline-container"
    />
  </div>,
  <footer>
    <p>
      hey, just chill..  ( ͡° ͜ʖ ͡°)
    </p>
  </footer>,
]
`;
